OSDOCS-4524: last minute 4.12 RN clean up (#54694)
openshift/openshift-docs: release_notes/ocp-4-12-release-notes.adoc

You must use {op-system} machines for the control plane, and you can use either {op-system} or {op-system-base} for compute machines.
//Removed the note per https://issues.redhat.com/browse/GRPA-3517

//TODO: Remove this for 4.13

> Starting with {product-title} {product-version} an additional six months of Extended Update Support (EUS) phase on even numbered releases from 18 months to 2 years. For more information, see the link:https://access.redhat.com/support/policy/updates/openshift[Red Hat {product-title} Life Cycle Policy].

Reviewer: ```suggestion
Starting with {product-title} {product-version} an additional six months of Extended Update Support (EUS) phase on even numbered releases from 18 months to two years. For more information, see the link:https://access.redhat.com/support/policy/updates/openshift[Red Hat {product-title} Life Cycle Policy].
```

[id="ocp-4-12-rhv"]
=== Red Hat Virtualization (RHV)

> This release provides several updates to Red Hat Virtualization (RHV). With this release

Reviewer: ```suggestion
This release provides several updates to Red Hat Virtualization (RHV). With this release:
```

[.small]
1. For more information about pod security admission in {product-title}, see xref:../authentication/understanding-and-managing-pod-security-admission.adoc#understanding-and-managing-pod-security-admission[Understanding and managing pod security admission].

[id="ocp-4-12-empty-file-support-removed"]
==== oc registry command for empty file support

> `--registry-config` or `--to option` in the `oc registry login` command will stop accepting empty files (non existing files will still be valid use case) and writing to `-`(stdout) is also removed.

Reviewer: ```suggestion
The `--registry-config` and `--to option` options in the `oc registry login` command will stop accepting empty files. These options will continue to work with files that do not exist. The ability to write output to `-`(stdout) is also removed.
```

> 

Author: FYI @bergerhoffer, @kalexand-rh, @JoeAldinger

Reviewer 2: For context: https://issues.redhat.com/browse/OSDOCS-4418

[discrete]
[id="ocp-4-12-dev-console-bug-fixes"]
==== Developer Console

* Previously, the `openshift-config` namespace was hard coded for the `HelmChartRepository` custom resource, which was the same namespace for the `ProjectHelmChartRepository` custom resource. This prevented users from adding private `ProjectHelmChartRepository` custom resources in their desired namespace. Consequently, users were unable to access secrets and configmaps in the `openshift-config` namespace. This update fixes the `ProjectHelmChartRepository` custom resource definition with a `namespace` field that can read the secret and configmaps from a namespace of choice by a user with the correct permissions. Additionally, the user can add secrets and configmaps to the accessible namespace, and they can add private Helm chart repositories in the namespace used the creation resources. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2071792[*BZ#2071792*])

[discrete]
[id="ocp-4-12-image-registry-bug-fixes"]
==== Image Registry

> * Previously, the image trigger controller did not have permissions to change objects. Consequently, image trigger annotations did not work on some resources. This update creates cluster role binding that provides the controller the required permissions to update objects according to annotations. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2055620[*BZ#2055620*])

Reviewer: ```suggestion
* Previously, the image trigger controller did not have permissions to change objects. Consequently, image trigger annotations did not work on some resources. This update creates a cluster role binding that provides the controller the required permissions to update objects according to annotations. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2055620[*BZ#2055620*])
```

> * Previously, the Image Registry Operator did not have a `progressing` condition for the `node-ca` daemon set and used `generation` from an incorrect object. Consequently, the `node-ca` daemon set could be marked as `degraded` while the Operator was still running. This update adds the `progressing` condition, which indicates that the installation is not complete. As a result, the Image Registry Operator successfully installs the `node-ca` daemon set and the installer waits until it is fully deployed. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2093440[[*BZ#2093440*])

Reviewer: ```suggestion
* Previously, the Image Registry Operator did not have a `progressing` condition for the `node-ca` daemon set and used `generation` from an incorrect object. Consequently, the `node-ca` daemon set could be marked as `degraded` while the Operator was still running. This update adds the `progressing` condition, which indicates that the installation is not complete. As a result, the Image Registry Operator successfully installs the `node-ca` daemon set, and the installer waits until it is fully deployed. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2093440[[*BZ#2093440*])
```

* Previously, when installing a cluster on {rh-openstack-first} with multiple networks defined in the `machineNetwork` parameter, the installation program only created security group rules for the first network. With this update, the installation program creates security group rules for all networks defined in the `machineNetwork` so that users no longer need to manually edit security group rules after installation. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2095323[*BZ#2095323*])

* Previously, users could manually set the API and Ingress virtual IP addresses to values that conflicted with the allocation pool of the DHCP server when installing a cluster on OpenStack. This could cause the DHCP server to assign one of the VIP addresses to a new machine, which would fail to start. In this update, the installation program validates the user-provided VIP addresses to ensure that they do not conflict with any DHCP pools. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1944365[*BZ#1944365*])

> * Previously, when installing a cluster on vSphere using a data center that is embedded inside a folder, the installation program could not locate the data center object, causing the installation to fail. In this update, the installation program can traverse the directory that contains the data center object, allowing the installation to succeed. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2097691[*BZ#2097691*])

Reviewer: "datacenter" is correct for vSphere.

* Previously, when installing a cluster on Azure using arm64 architecture with installer-provisioned infrastructure, the image definition resource for `hyperVGeneration` V1 incorrectly had an architecture value of `x64`. With this update, the image definition resource for `hyperVGeneration` V1 has the correct architecture value of `Arm64`. (link:https://issues.redhat.com/browse/OCPBUGS-3639[*OCPBUGS-3639*])

* Previously, when installing a cluster on VMware vSphere, the installation could fail if the user specified a user-defined folder in the `failureDomain` section of the `install-config.yaml` file. With this update, the installation program correctly validates user-defined folders in the `failureDomain` section of the `install-config.yaml` file. (link:https://issues.redhat.com/browse/OCPBUGS-3343[*OCPBUGS-3343*])

> * Previously, when destroying a partially deployed cluster after an installation failed on VMware vSphere, some virtual machine folders were not destroyed. This error could occur in clusters configured with multiple vSphere data centers or multiple vSphere clusters. With this update, all installer-provisioned infrastructure is correctly deleted when destroying a partially deployed cluster after an installation failure. (link:https://issues.redhat.com/browse/OCPBUGS-1489[*OCPBUGS-1489*])

Reviewer: "datacenter" is correct for vSphere.

* Previously, the secondary scheduler deployment was not deleted after a secondary scheduler custom resource was deleted. Consequently, the Secondary Schedule Operator and Operand were not fully uninstalled. With this update, the correct owner reference is set in the secondary scheduler custom resource so that it points to the secondary scheduler deployment. As a result, secondary scheduler deployments are deleted when the secondary scheduler custom resource is deleted. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2100923[*BZ#2100923*])

> * For the {product-title} {product-version} release, the descheduler can now publish events to an API group, because the release adds additional role-based access controls (RBAC) rules to the descheduler's profile.(link:https://issues.redhat.com/browse/OCPBUGS-2330[*OCPBUGS-2330*])

Reviewer: ```suggestion
* For the {product-title} {product-version} release, the descheduler can now publish events to an API group because the release adds additional role-based access controls (RBAC) rules to the descheduler's profile.(link:https://issues.redhat.com/browse/OCPBUGS-2330[*OCPBUGS-2330*])
```

* The `package-server-manifest` (PSM) is a controller that ensures that the correct `package-server` Cluster Service Version (CSV) is installed on a cluster. Previously, changes to the `package-server` CSV were not being reverted because of a logical error in the reconcile function in which an on-cluster object could influence the expected object. Users could modify the `package-server` CSV and the changes would not be reverted. Additionally, cluster upgrades would not update the YAML for the `package-server` CSV. With this update, the expected version of the CSV is now always built from scratch, which removes the ability for an on-cluster object to influence the expected values. As a result, the PSM now reverts any attempts to modify the `package-server` CSV, and cluster upgrades now deploy the expected `package-server` CSV. (link:https://issues.redhat.com/browse/OCPBUGS-858[*OCPBUGS-858*])

> * Previously, OLM would upgrade an Operator according to the Operator's CRD status. A CRD lists component references in an order defined by the group/version/kind (GVK) identifier. Operators that share the same components might cause the GVK to change the component listings for an Operator, and this can cause the OLM to require more system resources to continuously update the status of a CRD. With this update, the Operator Lifecycle Manager (OLM) now upgrades an Operator according to the Operator's component references. A change to the custom resource definition (CRD) status of an Operator does not impact the OLM Operator upgrade process.(link:https://issues.redhat.com/browse/OCPBUGS-3795[*OCPBUGS-3795*])

Reviewer: group/version/kind is a term of art, right? Otherwise, slashes between text isn't allowed.

Author: Correct, here is one example I see https://docs.openshift.com/container-platform/4.10/operators/operator_sdk/helm/osdk-helm-project-layout.html

* Before this update, users could not silence alerts in the *Developer* perspective in the {product-title} web console when a user-defined Alertmanager service was deployed because the web console would forward the request to the platform Alertmanager service in the `openshift-monitoring` namespace. With this update, when you view the *Developer* perspective in the web console and try to silence an alert, the request is forwarded to the correct Alertmanager service. (link:https://issues.redhat.com/browse/OCPBUGS-1789[*OCPBUGS-1789*])

> * Previously, there was a known issue in the *Add Helm Chart Repositories* form to extend the Developer Catalog of a project. The *Quick Start* guides shows that you can add the `ProjectHelmChartRepository` CR in the desired namespace whereas it does not mention that to perform this you need permission from the kubeadmin. This issue was resolved with *Quickstart* mentioning the correct steps to create `ProjectHelmChartRepository` CR. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2057306[*BZ#2057306*])

Reviewer: ```suggestion
* Previously, there was a known issue in the *Add Helm Chart Repositories* form to extend the Developer Catalog of a project. The *Quick Start* guides shows that you can add the `ProjectHelmChartRepository` CR in the required namespace whereas it does not mention that to perform this you need permission from the kubeadmin. This issue was resolved with *Quickstart* mentioning the correct steps to create `ProjectHelmChartRepository` CR. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2057306[*BZ#2057306*])
```

> * Previously, there was a known issue in the *Add Helm Chart Repositories* form to extend the Developer Catalog of a project. The *Quick Start* guides shows that you can add the `ProjectHelmChartRepository` CR in the desired namespace whereas it does not mention that to perform this you need permission from the kubeadmin. This issue was resolved with *Quickstart* mentioning the correct steps to create `ProjectHelmChartRepository` CR. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2057306[*BZ#2057306*])

Reviewer: ```suggestion
* Previously, there was a known issue in the *Add Helm Chart Repositories* form to extend the Developer Catalog of a project. The *Quick Start* guides shows that you can add the `ProjectHelmChartRepository` CR in the required namespace whereas it does not mention that to perform this you need permission from the kubeadmin. This issue was resolved with *Quickstart* mentioning the correct steps to create `ProjectHelmChartRepository` CR. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2057306[*BZ#2057306*])
```
